internal/handlers/images.go

```diff
@@ -59,12 +63,31 @@ func NewImageHandler(is imagestore.ImageStore, assistedServiceScheme, assistedSe
 		Recorder: metrics.NewRecorder(metricsConfig),
 	})
 
+	client := http.DefaultClient
```

Comment:
`DefaultClient` is actually an instantiated client. I think it would be better to create a new client rather than editing what is effectively global state.